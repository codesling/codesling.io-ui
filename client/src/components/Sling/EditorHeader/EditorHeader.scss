@import '../../../styles/vars';
@import '../../../styles/modules/mq';
@import '../../../styles/modules/colors';

// keep the total height 80px; CodeMirror height accounts for 80px header
$navbarHeight: 77px;

.editor-header {
  height: $navbarHeight;
  border-bottom: 3px solid palette('red');
  display: grid;
  grid-template-areas:
    "logo"
    "navbar";

  @include mq('md') {
    grid-template-areas:
      "logo navbar navbar";
    grid-template-columns:
      33% 33% 33%;
  }
}

.logo-container {
  grid-area: logo;
  display: flex;

  .logo {
    margin: auto;
    width: 200px;
    cursor: pointer;
    @include mq('md') {
      margin-left: $gutter;
    }
  }
}

.navbar-container {
  grid-area: navbar;
}

.editor-navbar {
  text-align: right;
  height: 100%;
  display: flex;
  
  ul {
    display: block;
    padding: 0;
    margin: auto;

    @include mq('md') {
      margin-right: $gutter;
    }
  }

  ul li {
    display: inline;
    margin-left: 30px;
    cursor: pointer;
  }
}
