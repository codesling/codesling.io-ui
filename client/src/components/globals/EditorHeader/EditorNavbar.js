import React from 'react';
import Logout from '../../Auth/Logout.jsx'
import Button from '../Button';

const EditorNavbar = (props) => (
  <nav className="editor-navbar">
    <ul>
      <li onClick={() => props.history.push('/challenge')}>My Challenges</li>
      <li onClick={() => props.history.push('/history')}>My History</li>
      <Logout history={props.history}/>
    </ul>
  </nav>
);

export default EditorNavbar;
