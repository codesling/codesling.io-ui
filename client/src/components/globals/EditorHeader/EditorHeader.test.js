import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import EditorHeader from './index';

test('EditorHeader Smoke Test', () => {
  const component = shallow(
    <EditorHeader
      text="Hello, world!"
    />
  );
  expect(toJSON(component)).toMatchSnapshot();
});
