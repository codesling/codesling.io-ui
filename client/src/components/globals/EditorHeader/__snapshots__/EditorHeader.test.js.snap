// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorHeader Smoke Test 1`] = `
<div
  className="editor-header"
>
  <div
    className="logo-container"
  >
    <Logo />
  </div>
  <div
    className="navbar-container"
  >
    <EditorNavbar />
  </div>
</div>
`;
